### PR TITLE
Null components [Fixes #17]

### LINKED ISSUE #17: Components returning null with different results for render and shallow
I came across an interesting inconsistency, let me demonstrate this with some code, here's a simplified version of my `ContentToggle` component:

``` js
export default class ContentToggle extends React.Component {
  props: {
    component: Function|string,
    content: ?ReactElement,
    options: ?Array<ReactElement>,
  };

  render(): ?ReactElement {
    const {content, options, ...otherProps} = this.props;

    if (!content && !options) {
      return null;
    }

    return React.createElement('a', { href: '#', ...otherProps });
  }
}
```

Important to note is that I either need to pass on `content` or `options` as props or otherwise it will return `null`.

My initial tests I wrote:

``` js
describe(ContentToggle.displayName, () => {
  it('renders children if content is passed', () => {
    const children = 'FOO';
    const el = render(<ContentToggle content={<span />}>{children}</ContentToggle>);

    expect(el).to.have.text(children);
  });

  it('renders children if options are passed', () => {
    const children = 'FOO';
    const el = render(<ContentToggle options={[]}>{children}</ContentToggle>);

    expect(el).to.have.text(children);
  });

 it('renders nothing if no content or options are passed', () => {
    const children = 'FOO';
    const el = render(<ContentToggle>{children}</ContentToggle>);

    expect(el).to.not.have.text(children);
    expect(el).to.be.blank();
  });
});
```

The first two tests passed, but the third one did not:

```
  1) undefined renders nothing if no content or options are passed:
     AssertionError: expected the node in <??? /> to be empty

     HTML:

     <noscript></noscript>
```

When I change the 3rd test to use `shallow` vs `render` this test passes, or I can write the test as follow:

``` js
describe(ContentToggle.displayName, () => {
  it('renders nothing if no content or options are passed', () => {
    const children = 'FOO';
    const el = render(<ContentToggle>{children}</ContentToggle>);

    expect(el).to.not.have.text(children);
    expect(el.find('noscript')).to.be.blank();
  });
});
```

I would have expected `expect(el).to.be.blank();` to work with both `render` as with `shallow` (or `mount` for that matter), or at least consistent behaviour across all 3.

After discussing this in https://github.com/airbnb/enzyme/issues/139 we've decided to tackle this within `chai-enzyme`.

### PR DESCRIPTION
**Pull request by ayrton**
Fixes #17

### CODE REVIEW

File: src/TestWrapper.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 export default class TestWrapper {
   isPresent () {
-    return this.wrapper.length > 0
+    return this.wrapper.children().length > 0
```
Comment (ayrton) [inline src/TestWrapper.js]: This is probably not be what we want, I was confused initially that the behaviour of `isPresent` & `isEmpty` was different.

**Comment by vesln** [comment]
:+1:

**Comment by vesln** [comment]
worth mentioning in the readme too

**Comment by ayrton** [comment]
@vesln good call, will add examples for `present` & `blank` in the readme

File: README.md [hunk 2]
```diff
@@ -307,6 +307,17 @@ expect(wrapper.find('#parent')).to.not.be.blank()
...
 expect(wrapper.find('#child')).to.be.empty // an alias
 expect(wrapper.find('#parent')).to.not.be.empty // an alias
+
+class NullFixture extends React.Component {
+  render () {
+    return null
+  }
+}
+
+const nullWrapper = moun(<NullFixture />) // mount/render/shallow when applicable
```
Comment (andreasklinger) [inline README.md]: typo

**Comment by andreasklinger** [comment]
:+1:

**Pull request merged**